### PR TITLE
Specify remote bucket/collection in sync (fixes #451)

### LINKED ISSUE #451: Collection#sync should let you specify a different remote collection
Per https://bugzilla.mozilla.org/show_bug.cgi?id=1276935, we want to avoid leaking information by syncing to a collection whose name is too transparent, but we don't always have access to what the name "should" be. It would be nice if we could synchronize to a remote collection whose name was not the same as the local collection. Perhaps this could be specified as an argument to `Collection#sync`.

### PR DESCRIPTION
**Pull request by leplatrem**
@glasserc r?

### CODE REVIEW

**Comment by glasserc** [comment]
The tests seem to be failing, although I can't figure out why, but it does seem related to this change.

Can we just create one reference to the local collection and pass that around to `pushChanges` and `pullChanges` instead of giving them `collection` and `bucket` options? I think that would be cleaner.

File: test/collection_test.js [hunk 1]
```diff
@@ -2200,14 +2185,14 @@ describe("Collection", () => {
       it("should transfer the headers option", () => {
         return articles.sync({headers: {Foo: "Bar"}})
           .then(() => {
-            expect(pullChanges.firstCall.args[1]).eql({headers: {Foo: "Bar"}});
+            expect(pullChanges.firstCall.args[1]).to.have.property("headers").eql({Foo: "Bar"});
```
Comment (glasserc) [inline test/collection_test.js]: Why are these changes necessary? (Although they look OK to me)

Comment (leplatrem) [inline test/collection_test.js]: If the options have extra entries (defaut values or whatever unreleated to the test description here) they have to be mentionned. With this change we test only a subpart of the specified options, the ones concerned by the test (ie. headers here)

Comment (glasserc) [inline test/collection_test.js]: Ah, I understand, it's because we're adding options to the `pullChanges` calls to track the collection we're syncing against.

**Comment by glasserc** [comment]
OMG, I just noticed you updated this a while ago and I haven't rereviewed it. Looks good now. Merging.
